docs/src/constraints.md

```diff
@@ -1,5 +1,510 @@
-Constraints
-===========
+```@meta
+CurrentModule = JuMP
+DocTestSetup = quote
+    using JuMP
+    const MOI = JuMP.MathOptInterface
+end
+DocTestFilters = [r"≤|<=", r"≥|>=", r" == | = ", r" ∈ | in "]
+```
+
+# Constraints
+
+This page explains how to write various types of constraints in JuMP. Before
+reading further, please make sure you are familiar with JuMP models, and JuMP
+[Variables](@ref). If you want to add nonlinear constraints, read
+[Nonlinear Modeling](@ref) instead.
+
+JuMP is based on the MathOptInterface API. Because of this, JuMP thinks of a
+constraint as a *function* belonging to a *set*. For example, instead of
+thinking  about a constraint ``a^\top x \le b`` as a *less-than-or-equal-to*
+constraint, JuMP thinks about this as the *scalar affine* function ``a^\top x``
+belonging to the *less-than* set ``(-\infty, b]``. Thus, instead of a
+*less-than-or-equal-to* constraint, we consider this constraint to be a  *scalar
+affine -in- less than* constraint. More generally, we use the shorthand
+*function-in-set* to refer to constraints composed of different types of
+functions and sets. In the rest of this page, we will introduce the different
+types of functions and sets that JuMP knows about as needed. You can also more
+details about this *function-in-set* concept in the MathOptInterface
+documentation.
+
+!!! note
+    Throughout this page (and these docs), we use `MOI` as a shorthand for the
+    `MathOptInterface` module. This can be created by including the line `const
+    MOI = JuMP.MathOptInterface` after `using JuMP` in your code.
```

Comment:
Why not `MOI = JuMP.MOI` ? People will prefer something shorter

Reply:
Updated. But I have a question. What happens if JuMP uses a different version of MOI to the one the user gets from `using MathOptInterface`? Can they interact?

Reply:
How is it possible for JuMP to use a different version of MOI than the one from `using MathOptInterface`?

Reply:
Hmm, for some reason I thought Pkg3 allowed JuMP to depend upon a different version of a package compared to the one I add. Never mind.